package.json

```diff
@@ -67,6 +67,13 @@
         "command": "ocaml.switch-impl-intf",
         "key": "Alt+O",
         "when": "editorLangId  == ocaml || editorLangId  == ocaml.interface || editorLangId == reason || editorLangId == ocaml.ocamllex || editorLangId == ocaml.menhir"
+      },
+      {
+        "command": "editor.action.codeAction",
+        "key": "Alt+D",
```

Comment:
Should this binding be available in mli files?

Reply:
Doesn't the `destruct` code action only expand to `match ... with` expressions? That wouldn't be valid in a mli. The language server doesn't even provide a `destruct` code action in an mli.

Reply:
Indeed. As far as I can see we aren't checking for `editorLangId == ocaml || editorLangId == reason` in this binding. Does that mean it will be available in mli, ocamllex, and menhir files?

Reply:
Oh, I see your point. That is indeed a problem. Using `Alt+D` in any file - even non-OCaml files  will error with `No code actions for destruct available`.

Reply:
I pushed a commit that fixes this with the condition you mentioned.